src/snap/utils.ts

```diff
@@ -17,3 +17,20 @@ export function isMetaMaskErrorObject(e: unknown): boolean {
   if (!("originalError" in e["data"])) return false;
   return true;
 }
+
+export function isElementVisible(
+  page: DappeteerPage,
+  selector: string,
+  timeout = 2000
```

Comment:
2 sec seems a lot for checking if an element is visible

Reply:
Agreed, only 1 second was mostly not enough the 2 times I used this function. I can have a 1s be the default and still use 2 sec in my particular test.

The very good part about this, is that after all it'll be a _maximum_ 2s so if the element is there in less time, this function will actually take less. 


```suggestion
  timeout = 1000
```

Reply:
Right haha I knew before! it's waiting max 2s, but I think it makes the api so much cleaner that  way!

Reply:
Thanks to Bero's push, we came up with something even more elegant. I'm now racing between 2 possible elements that could appear, the issue of the race telling us in which state we are in. It's relatively elegant beside the bad name I came up with.